### PR TITLE
Copied EncodedKey and RawKeyValue to storage/engine - Issue #2925

### PR DESCRIPTION
**Pull request by vladdy**
I copied EncodedKey and RawKeyValue to storage/engine leaving original types from roachpb in place. Tests pass. Please let me know if there are better destination files for my changes or whenever I missed something. I'll be happy to fix that.

Also, the issue mentions a clarification for those types. Any suggestions on what can help to get an idea about that? Since I'm very new to the project, I'm not confident on how to proceed with this.

[Review on Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/2979)

### CODE REVIEW

**Comment by tamird** [comment]
Let's leave the clarifications to another change. In the meantime, can you remove these types from roachpb? We would rather not have them in both places.

File: storage/engine/data.go [hunk 1]
```diff
@@ -0,0 +1,51 @@
+// Copyright 2014 The Cockroach Authors.
```
Comment (tamird) [inline storage/engine/data.go]: can you rename this file to `encoded_key.go`?

File: util/keys/bytes.go [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+// implied. See the License for the specific language governing
+// permissions and limitations under the License. See the AUTHORS file
+// for names of contributors.
+//
+// Author: Spencer Kimball (spencer.kimball@gmail.com)
+
+package keys
```
Comment (tamird) [inline util/keys/bytes.go]: we already have a package named keys, albeit with a different path. Still, this is confusing.

File: storage/engine/data.go (hunk 1 above)
Comment (tamird) [inline storage/engine/data.go]: on second thought, i think you can put this in mvcc.go and rename the type to `MVCCKey` as suggested by @spencerkimball

**Comment by petermattis** [comment]
Thanks for tackling this. Definitely remove `EncodedKey` and `RawKeyValue` from `roachpb`. 


---

Review status: 0 of 23 files reviewed at latest revision, 6 unresolved discussions.

---

<sup>**[roachpb/data.go, line 102 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/2979#-K27VxZw2ViubN-5o1ZX)** ([raw file](https://github.com/cockroachdb/cockroach/blob/202cdf20297c503eb8e349dbae9197c055376c41/roachpb/data.go#L102)):</sup>
You should remove this type. Looks like you've already removed all references to it.


---

<sup>**[roachpb/data.go, line 114 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/2979#-K27W1cc5ODNykFQGy4d)** ([raw file](https://github.com/cockroachdb/cockroach/blob/9aa58f05e6cba1d5b25eed1c16fe0d4caaf9e390/roachpb/data.go#L114)):</sup>
I'm not sure it is worth trying to share this method between `roachpb` and `storage/engine`. A little bit of code duplication is not harmful. If you're strongly in the duplication-of-code-is-harmful-at-all-times camp, then just leave the method here, but export it by renaming to `BytesNext`.


---

<sup>**[storage/engine/data.go, line 31 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/2979#-K27WUKMXemnG_s4KbYR)** ([raw file](https://github.com/cockroachdb/cockroach/blob/202cdf20297c503eb8e349dbae9197c055376c41/storage/engine/data.go#L31)):</sup>
Let's rename to `MVCCKey` as per the discussion in #2925.


---

<sup>**[storage/engine/mvcc.proto, line 87 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/2979#-K27Wk0b6q6zvY4DiRXU)** ([raw file](https://github.com/cockroachdb/cockroach/blob/202cdf20297c503eb8e349dbae9197c055376c41/storage/engine/mvcc.proto#L87)):</sup>
Let's rename to `MVCCKeyValue`.


---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/2979)

**Comment by vladdy** [comment]
Thanks for the feedback. I'll work on the corrections later this week.

**Pull request closed without merging**